misk/src/main/kotlin/misk/hibernate/HibernateHealthCheck.kt

```diff
@@ -0,0 +1,56 @@
+package misk.hibernate
+
+import misk.healthchecks.HealthCheck
+import misk.healthchecks.HealthStatus
+import misk.jdbc.DataSourceConfig
+import org.hibernate.PessimisticLockException
+import org.hibernate.QueryTimeoutException
+import org.hibernate.SessionFactory
+import java.sql.Timestamp
+import java.time.Clock
+import java.time.Duration
+import javax.inject.Singleton
+import javax.persistence.LockTimeoutException
+import javax.persistence.NoResultException
+import javax.persistence.NonUniqueResultException
+import javax.persistence.PersistenceException
+import javax.transaction.TransactionRequiredException
+
+/**
+ * HealthCheck to confirm database connectivity and defend against clock skew.
+ */
+class HibernateHealthCheck(
+  private val sessionFactory: SessionFactory,
+  private val config: DataSourceConfig,
+  private val clock: Clock
+) : HealthCheck {
+
+  override fun status(): HealthStatus {
+    val databaseInstant = try {
+      sessionFactory.openSession().use { session ->
+        session.createNativeQuery("SELECT NOW()").uniqueResult() as Timestamp
+      }.toInstant()
+    } catch (e: Exception) {
+      return when (e) {
+        is NoResultException,
+        is NonUniqueResultException,
+        is IllegalStateException,
+        is QueryTimeoutException,
+        is TransactionRequiredException,
+        is PessimisticLockException,
+        is LockTimeoutException,
+        is PersistenceException ->
+          HealthStatus.unhealthy("Hibernate: failed to query ${config.database}")
```

Comment:
why not report all exceptions as unhealthy?

Reply:
It felt bad catching `Exception`. These are the specific exceptions thrown by an API that I'm no longer calling. woops. Will remove.